main/adv_post.c

```diff
@@ -299,5 +295,8 @@ adv_post_init(void)
     adv_reports.num_of_advs = 0;
     api_callbacks_reg((void *)&adv_callback_func_tbl);
     const uint32_t stack_size = 1024U * 4U;
-    xTaskCreate(&adv_post_task, "adv_post_task", stack_size, NULL, 1, NULL);
+    if (!os_task_create(&adv_post_task, "adv_post_task", stack_size, NULL, 1, NULL))
+    {
+        LOG_ERR("Can't create thread");
```

Comment:
Some of the task creation failures are fatal, for example failing here means gateway is functional. What would be appropriate action to take? Reboot? Erase configured settings and reboot? Blink error through LEDs and let user figure it out?